src/components/Tooltip/Tooltip.tsx

```diff
@@ -657,13 +639,21 @@ const Tooltip = ({
         styles[variant],
         className,
         `react-tooltip__place-${actualPlacement}`,
-        {
-          'react-tooltip__show': canShow,
-          [coreStyles['show']]: canShow,
-          [coreStyles['fixed']]: positionStrategy === 'fixed',
-          [coreStyles['clickable']]: clickable,
-        },
+        coreStyles[canShow ? 'show' : 'closing'],
+        canShow ? 'react-tooltip__show' : 'react-tooltip__closing',
+        positionStrategy === 'fixed' && coreStyles['fixed'],
+        clickable && coreStyles['clickable'],
       )}
+      onTransitionEnd={(event: TransitionEvent) => {
+        /**
+         * @warning if `--rt-transition-closing-delay` is set to 0,
+         * the tooltip will be stuck (but not visible) on the DOM
+         */
+        if (show || event.propertyName !== 'opacity') {
+          return
+        }
+        setRendered(false)
+      }}
```

Comment:
Is there an obvious alternative to this? Thought about doing a check on the `--rt-transition-closing-delay` variable to setup a timeout for `setRendered(false)` (as it was being done before, see code removed above), but it felt very hacky.

Reply:
I do agree, we can find a better solution for this on V6